backend/handler/question_groups.go

```diff
@@ -28,6 +28,30 @@ func (s *Server) GetQuestionGroups(e echo.Context) error {
 	return e.JSON(http.StatusOK, res)
 }
 
+func (s *Server) GetQuestionGroup(e echo.Context, questionGroupID int, params GetQuestionGroupParams) error {
+    questionGroup, err := s.repo.GetQuestionGroup(uint(questionGroupID))
+
+    if err != nil {
+        e.Logger().Errorf("failed to get question group: %v", err)
+
+        return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+    }
+
+    if questionGroup == nil {
+        return echo.NewHTTPError(http.StatusNotFound, "Not found")
+    }
```

Comment:
存在しない質問グループを取得しようとすると、`questionGroup == nil`のところに到達する前にGORMの方でNot foundエラーになります。`if err != nil`のブロックの中で判定するのが良いと思います
```suggestion
    if err != nil {
        if errors.Is(err, model.ErrNotFound) {
            return echo.NewHTTPError(http.StatusNotFound, "Not found")
        }

        e.Logger().Errorf("failed to get question group: %v", err)

        return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
    }
```

Reply:
firstなのにfindだと思ってコードを書いてしまってました。修正します！